config/confighttp/confighttp.go

```diff
@@ -0,0 +1,48 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package confighttp
+
+import (
+	"net/http"
+	"time"
+
+	"go.opentelemetry.io/collector/config/configtls"
+)
+
+type HttpClientSettings struct {
+	// The target to which the exporter is going to send traces or metrics,
+	// using the gRPC protocol. The valid syntax is described at
+	// https://github.com/grpc/grpc/blob/master/doc/naming.md.
+	Endpoint string `mapstructure:"endpoint"`
+
+	// TLSSetting struct exposes TLS client configuration.
+	TLSSetting configtls.TLSClientSetting `mapstructure:",squash"`
+
+	// Timeout parameter configures `http.Client.Timeout`.
+	Timeout time.Duration `mapstructure:"timeout,omitempty"`
+}
+
+func (hcs *HttpClientSettings) ToClient() (*http.Client, error) {
+	tlsCfg, err := hcs.TLSSetting.LoadTLSConfig()
+	if err != nil {
+		return nil, err
+	}
+	return &http.Client{
+		Transport: &http.Transport{
```

Comment:
Instantiating http.Transport directly like this isn't safe unless you give defaults for all the various values. e.g. IdleConnTimeout zero value defaults to no timeout.

See https://github.com/signalfx/signalfx-agent/blob/master/pkg/core/common/httpclient/http.go#L54 on how to use default transport values.

Reply:
Done.